### PR TITLE
Implement Value::is_unordered_link

### PR DESCRIPTION
**Pull request by ngeiswei**
Addresses https://github.com/opencog/atomspace/issues/2423

### CODE REVIEW

**Comment by ngeiswei** [comment]
The following unit test

https://github.com/opencog/atomspace/pull/2442/commits/efedf3f45d20054cd493c1537a5a39877cad0734

fails.

As far as I can tell, the factory for the `UnorderedLink` class is never called. Instead the factory for `EvaluationLink` is called because `EqualLink` inherits `VirtualLink` which (oddly) inherits `EvaluationLink`.

So `EqualLink` has multiple inheritances, `UnorderedLink` and `EvaluationLink`, aren't multiple factories supposed to be called?

**Comment by ngeiswei** [comment]
I can confirm that the `UnorderedLink` ctor never gets called.

Looking `ClassServer::splice` it's clear `ClassServer::_atomFactory` only associates one factory per type. That seems like a deficiency of our factory system. Not sure where to go from there. Should we auto generate multi-inherited classes (such as `EqualLink`) when needed, as to "simulate" multi-factory?

**Comment by linas** [comment]
Change to `VIRTUAL_LINK <- EVALUATABLE_LINK`, I think this would solve the problem.

Re; "Multiple factories", I want to avoid  c++ multiple inheritance .... at all costs, even. Down that path lies madness.   The Atom multiple inheritance is used primarily (only) to firm up the expected input and output types of the atoms, to firm up run-time type checking.

**Comment by ngeiswei** [comment]
`VIRTUAL_LINK <- EVALUATABLE_LINK` works!

If we're dedicated to avoid multi-factory, then I can let the implementation of `is_unordered_link` as is it, otherwise, I would rather replace it with

```c++
virtual bool is_unordered_link() const
{
    return nameserver().isA(get_type(), UNORDERED_LINK);
}
```

which would be more resilient to factory f-up.

File: opencog/atoms/core/FreeLink.cc [hunk 1]
```diff
@@ -57,7 +57,7 @@ FreeLink::FreeLink(const Link& l)
 
 void FreeLink::unorder(void)
 {
-	if (not nameserver().isA(get_type(), UNORDERED_LINK)) return;
+	if (not is_unordered_link()) return;
```
Comment (linas) [inline opencog/atoms/core/FreeLink.cc]: !!?? But FreeLinks are always ordered! how dd we end up with this?

**Pull request merged**